src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -340,6 +353,62 @@ public CourseConfirmState getConfirmState(DocumentReference regDocRef) {
     return confirmState;
   }
 
+  @Override
+  public List<DocumentReference> getRegistrationsForCourse(SpaceReference regSpaceRef,
+      List<String> sortFields) throws LuceneSearchException {
+    LuceneQuery query = searchService.createQuery();
+    query.add(searchService.createSpaceRestriction(regSpaceRef));
+    query.add(searchService.createObjectRestriction(getCourseParticipantClassRef()));
+    LuceneSearchResult result = searchService.search(query, sortFields, null);
+    return result.getResults(DocumentReference.class);
+  }
+
+  @Override
+  public List<DocumentReference> getRegistrationsForCourse(DocumentReference courseDocRef)
+      throws LuceneSearchException {
+    List<String> sortFields = new ArrayList<>();
+    return getRegistrationsForCourse(getRegistrationSpace(courseDocRef), sortFields);
+  }
+
+  @Override
+  public long getRegistrationCount(DocumentReference courseDocRef) throws LuceneSearchException {
+    return getRegistrationCount(courseDocRef, null);
+  }
+
+  @Override
+  public long getRegistrationCount(DocumentReference courseDocRef, CourseConfirmState state)
+      throws LuceneSearchException {
+    List<DocumentReference> announcementList = getRegistrationsForCourse(courseDocRef);
+
+    long retVal = 0;
+    for (EntityReference announcement : announcementList) {
+      DocumentReference announcementDocRef = new DocumentReference(announcement);
+      try {
+        List<BaseObject> partiObjs = modelAccess.getXObjects(announcementDocRef,
+            getCourseParticipantClassRef());
+        for (BaseObject obj : partiObjs) {
+          if (state == null) {
```

Comment:
if `state == null` there is no need to loop the objects, instead `retVal += partiObjs.size()`